### PR TITLE
 DAuth deliveries

### PR DESCRIPTION
**Pull request by yyd106**
# Milestone Delivery Checklist

- [x] The [milestone-delivery-template.md](https://github.com/w3f/Grant-Milestone-Delivery/blob/master/deliveries/milestone-delivery-template.md) has been copied and updated.
- [x] This pull request is being made by the same account as the accepted application.
- [x] I have disclosed any and all sources of reused code in the submitted repositories and have done my due diligence to meet its license requirements.
- [ ] In case of acceptance, an invoice must be submitted and the payment will be transferred to the Polkadot/fiat account provided in the application.
- [x] The delivery is according to the [Guidelines for Milestone Deliverables](https://grants.web3.foundation/docs/Support%20Docs/milestone-deliverables-guidelines).

Link to the application pull request: https://github.com/w3f/Grants-Program/pull/1655

### CODE REVIEW

**Comment by meldien** [comment]
@yyd106 Thank you for sending the invoice. I guess you did a tip mistake. In the note it's written you expecting 12'000 USDC but the due amount is way higher then this. 
Could you please double check and resend the invoice with the correct due amount?
Many thanks

**Comment by yyd106** [comment]
> @yyd106 Thank you for sending the invoice. I guess you did a tip mistake. In the note it's written you expecting 12'000 USDC but the due amount is way higher then this. Could you please double check and resend the invoice with the correct due amount? Many thanks

Hi @meldien , what do you mean by due amount, and where I can find it? The context is: We previously submitted the Keysafe project and then received the first Milestone Grants, 15,000 USDC. DAuth is our project post-pivot. This time we submitted two milestones, but the amount of this invoice only includes Milestone 2.

**Comment by meldien** [comment]
@yyd106 in your calculation it's written 100 X 12,000 = 1,200,000.00 USD
In my understanding you should change the 100 to 1x 12,000 = 12,000 USD. Then it will macht with the 12,000 USD in the comment.

**Comment by yyd106** [comment]
> @yyd106 in your calculation it's written 100 X 12,000 = 1,200,000.00 USD In my understanding you should change the 100 to 1x 12,000 = 12,000 USD. Then it will macht with the 12,000 USD in the comment.

It's my fault. Now fixed @meldien

**Comment by meldien** [comment]
@yyd106 no worries :) Thank you for the new invoice.

File: deliveries/dauth_network_milestone_1_2.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# Milestone Delivery :mailbox:
+
+> ⚡ Only the GitHub account that submitted the application is allowed to submit milestones. 
+> 
+> Don't remove any of the mandatory parts presented in bold letters or as headlines! Lines starting with `>`, such as this one, can be removed.
+
+**The delivery is according to the official [milestone delivery guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/milestone-deliverables-guidelines.md).**  
+
+* **Application Document:** https://github.com/w3f/Grants-Program/pull/1655
```
Comment (semuelle) [inline deliveries/dauth_network_milestone_1_2.md:9]:
```suggestion
* **Application Document:** https://github.com/w3f/Grants-Program/blob/master/applications/dauth_network.md
```

File: deliveries/dauth_network_milestone_1_2.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+
+**Deliverables**
+> Please provide a list of all deliverables of the milestone extracted from the initial application and a link to the deliverable itself. Ideally all links inside the below table should include a commit hash, which will be used for testing. If you don't provide a commit hash, we will work off the default branch of your repository. Thus, if you plan on continuing work after delivery, we suggest you create a separate branch for either the delivery or your continuing work. 
+> 
+> If there is anything particular about any of the deliverables we or a future reader should know, use the respective `Notes` column.
+
+**Milestone 1**
+
+| Number | Deliverable | Link | Notes |
```
Comment (PieWol) [inline deliveries/dauth_network_milestone_1_2.md:20]: Please change the milestone tables to match the ones declared in the application. See [here](https://github.com/w3f/Grants-Program/blob/5cc3d7198c45d0ea24c8694ff4047024a4f18ed7/applications/dauth_network.md?plain=1#L134C1-L145C1)

**Comment by PieWol** [comment]
Hey @yyd106,  I suggest to check the [FAQ entry](https://grants.web3.foundation/docs/faq#-milestone-delivery) on milestone deliveries again. It's possible to submit two milestones in one PR but please note that the milestones are only accepted together and can't be handled individually.

If you instead submit each milestone in a separate PR any delay or problem with a milestone doesn't hinder the others from being accepted and paid.

**Comment by yyd106** [comment]
Hi @PieWol , I have updated the milestones accordingly. Please check and let me know if any issue.

**Comment by PieWol** [comment]
Hey @yyd106 ,
I see you want to stick with two milestones in one PR. Thats fine. 
You have now mixed the tables of miletsone 1 and two. Please keep them separate and exactly as in your application. I linked the requested layout in my previous comment. Thank you.

**Comment by yyd106** [comment]
Hi @PieWol , 
I have separated the milestones. However, there are two points of clarification to add:

The test guide and SDK documentation are combined into a single document because the development iterates on the same product.
Regarding the item "Polkadot.js | Add in encryption/decryption functionality to @polkadot/keyring and @polkadot/extension", it is the deliverable of Keysafe. Therefore, it is not included in this delivery.

**Comment by PieWol** [comment]
Thanks @yyd106 the layouts look good now. 

Regarding the M1 deliverable number five (PolkadotJS) I wonder what your annotation means, stating that it is a deliverable by Keysafe and not in scope of the milestone. Would you mind explaining what that means, who Keysafe is and why it's out of scope? I couldn't find any mention of Keysafe in the application for this grant.

**Comment by yyd106** [comment]
Hi @PieWol , 
Keysafe was the predecessor of the DAuth project team. It also applied for Polkadot Grants and successfully submitted Milestone1. Here is the [Keysafe application](https://github.com/w3f/Grants-Program/blob/master/applications/keysafe_network.md)

Keysafe is a project that implements non-custodial private key management based on TEE + MPC technology. After discovering that there were not many opportunities for MPC wallets, we expanded the original technology using ZK to achieve decentralized verification and pushed into areas such as DID and Proof of Humanity. This is what is now known as DAuth Network.

**Comment by semuelle** [comment]
Hi @yyd106,

wasn't M1 - or at least significant parts of it - covered and paid for as [part of the Keysafe grant](https://github.com/w3f/Grant-Milestone-Delivery/pull/639)? 

> The test guide and SDK documentation are combined into a single document because the development iterates on the same product.

This is fine as long as the content is there and discoverable. You can link to the same document for two separate deliverables.

**Comment by PieWol** [comment]
Hey @yyd106 
I'm trying to run the unit tests on the dockerfiles in this [step](https://github.com/DAuth-Network/instant-proof/tree/polka-grant#inside-docker-instance-build-package) I get the following error. I'm executing this inside the instant-proof docker instance.

```
#   cd /root/incubator-teaclave-sgx-sdk/samplecode/instant-proof/
  make SGX_MODE=SW # 
/opt/sgxsdk//bin/x64/sgx_edger8r --trusted enclave/Enclave.edl --search-path /opt/sgxsdk//include --search-path /root/incubator-teaclave-sgx-sdk/edl --trusted-dir enclave
/opt/sgxsdk//bin/x64/sgx_edger8r --untrusted enclave/Enclave.edl --search-path /opt/sgxsdk//include --search-path /root/incubator-teaclave-sgx-sdk/edl --untrusted-dir app
GEN  =>  enclave/Enclave_t.c enclave/Enclave_t.h app/Enclave_u.c app/Enclave_u.h
CC   <=  enclave/Enclave_t.c
ar rcsD lib/libEnclave_u.a app/Enclave_u.o
/bin/sh: 1: cargo: not found
Makefile:115: recipe for target 'bin/app' failed
make: *** [bin/app] Error 127
```

Do you have any idea why strictly following your setup doesn't work or where I went wrong?

**Comment by ksrust** [comment]
Hi, @PieWol 

I was testing in my local environemnt, couldn't reproduce your issues.
Can you try these commands in your docker instance?

```
cat ~/.cargo/env 
source ~/.cargo/env
which cargo
```
And let me know the results.

**Comment by PieWol** [comment]
Hey @ksrust , the output is the following:
```
# cat ~/.cargo/env 
export PATH="$HOME/.cargo/bin:$PATH"
# source ~/.cargo/env
/bin/sh: 2: source: not found
# which cargo
```

**Comment by PieWol** [comment]
That was indeed the issue. Unit test were flawless on x86.

**Comment by ksrust** [comment]
Glad to hear it works. 
It is my mistake, I forget to mention docker image includes sgx-related features does not work on apple m1/m2. 
I'll update that in the doc.

**Comment by PieWol** [comment]
Hey @ksrust , 
very good idea to add it to the docs, appreciate it 👍.

I've already created an [evaluation](https://github.com/w3f/Grant-Milestone-Delivery/pull/1117) which is waiting for review itself.

**Comment by PieWol** [comment]
Hey @yyd106 and @ksrust ,
I'm sorry for the inconvenience but there are a few details that need clarification which I missed earlier. I updated my evaluation accordingly. Please review the [evaluation](https://github.com/w3f/Grant-Milestone-Delivery/pull/1117) again and post any clarification to this in this PR. Looking forward to your reply.

**Comment by yyd106** [comment]
> Hey @yyd106 and @ksrust , I'm sorry for the inconvenience but there are a few details that need clarification which I missed earlier. I updated my evaluation accordingly. Please review the [evaluation](https://github.com/w3f/Grant-Milestone-Delivery/pull/1117) again and post any clarification to this in this PR. Looking forward to your reply.

**Comment by yyd106** [comment]
Hi @PieWol 
I have updated the [evaluation](https://github.com/w3f/Grant-Milestone-Delivery/pull/1117) accordingly. Please let me know if anything else you need.

**Comment by PieWol** [comment]
hey @yyd106 
I have replied to you in my evaluation. pinging you here just to get your attention on this matter.

**Comment by yyd106** [comment]
Hi @PieWol , the comments has been updated accordingly and replied in the evaluation PR.

**Comment by RouvenP** [comment]
hi @yyd106 we just sent the payment

**Comment by yyd106** [comment]
@RouvenP Thanks, we confirm that we have received the payment.

**Pull request merged**